oauth2_provider/oauth2_validators.py

```diff
@@ -461,6 +485,24 @@ def get_code_challenge_method(self, code, request):
     def save_authorization_code(self, client_id, code, request, *args, **kwargs):
         self._create_authorization_code(request, code)
 
+    def get_authorization_code_scopes(self, client_id, code, redirect_uri, request):
+        scopes = []
+        fields = {
+            "code": code,
+        }
+
+        if client_id:
+            fields["application__client_id"] = client_id
+
+        if redirect_uri:
+            fields["redirect_uri"] = redirect_uri
```

Comment:
I don't see why we are filtering by these values here. `oauthlib` says:

```
        With that the token endpoint knows when to include OpenIDConnect
        id_token in token response only based on authorization code scopes.

        Only code param should be sufficient to retrieve grant code from
        any storage you are using, `client_id` and `redirect_uri` can have a
        blank value `""` don't forget to check it before using those values
        in a select query if a database is used.
```

This function is only used to allow `oauthlib` dispatcher to work out whether to use an `openid` dispatcher or fallback to `oauth2` - code should be sufficient, the other fields are validated later.